fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/PhysicalDataSourceConstraint.java

```diff
@@ -0,0 +1,127 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.table.resolver;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.table.PhysicalTableSchema;
+import com.yahoo.bard.webservice.web.ApiFilter;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * Data source constraint containing physical name of the columns.
+ */
+public class PhysicalDataSourceConstraint extends DataSourceConstraint {
+
+    private final Set<String> allColumnPhysicalNames;
+
+    /**
+     * Constructor.
+     *
+     * @param dataSourceConstraint  Data source constraint containing all the column names as logical names
+     * @param physicalTableSchema  A map from logical column name to physical column names
+     */
+    public PhysicalDataSourceConstraint(
+            @NotNull DataSourceConstraint dataSourceConstraint,
+            @NotNull PhysicalTableSchema physicalTableSchema
+    ) {
+        super(dataSourceConstraint);
+
+        Set<String> schemaColumnNames = physicalTableSchema.getColumnNames();
+
+        this.allColumnPhysicalNames = dataSourceConstraint.getAllColumnNames().stream()
+                .filter(schemaColumnNames::contains)
+                .map(physicalTableSchema::getPhysicalColumnName)
+                .collect(Collectors.collectingAndThen(Collectors.toSet(), Collections::unmodifiableSet));
+    }
+
+    /**
+     * Constructor, use with care, beware of caching behavior.
+     *
+     * @param dataSourceConstraint  Data source constraint containing all the column names as logical names
+     * @param allColumnPhysicalNames  The physical names of the columns
+     */
+    private PhysicalDataSourceConstraint(
+            @NotNull DataSourceConstraint dataSourceConstraint,
+            @NotNull Set<String> allColumnPhysicalNames
+    ) {
+        super(dataSourceConstraint);
+        this.allColumnPhysicalNames = allColumnPhysicalNames;
+
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param requestDimensions  Dimensions contained in request
+     * @param filterDimensions  Filtered dimensions
+     * @param metricDimensions  Metric related dimensions
+     * @param metricNames  Names of metrics
+     * @param allDimensions  Set of all dimension objects
+     * @param allDimensionNames  Set of all dimension names
+     * @param allColumnNames  Set of all column names
+     * @param apiFilters  Map of dimension to its set of API filters
+     * @param allColumnPhysicalNames  Set of all column physical names
+     */
+    protected PhysicalDataSourceConstraint(
```

Comment:
Why do we have this constructor? It looks like it does exactly the same thing as the private constructor just above, except unpacks the parameters. Can we remove this one?